jest.config.js

```diff
@@ -0,0 +1,11 @@
+module.exports = {
+  preset: 'ts-jest',
+  setupFilesAfterEnv: [
+    '<rootDir>/src/__test__/setupTests.ts',
+  ],
+  transform: {
+    '^.+\\.(js|jsx|ts|tsx)$': '<rootDir>/node_modules/babel-jest',
+  },
+  testEnvironment: 'jsdom',
```

Comment:
What do you think to add a   `testMatch` with `spec` identifier ['<rootDir>/tests/**/?(*.)+(spec).[jt]s?(x)']?

Reply:
@LeoSilvaGomes Just as it is necessary to use Babel to transpile the TypeScript code to JavaScript at build time so that it becomes compatible with browsers, Jest also needs to transpile the code with modern syntax to be able to run the tests becomes the tests are run inside NodeJS.
Therefore,  this transform property is needed to take all the files, whether from tests or not, and transform them into compatible code for the test execution. With that in mind, I don't believe it would be correct to add "spec" to the regex because we are not only transforming the test files but all the application files.

The link for the documentation explaining the use of this property can be found here: [Transform - Jest](https://jestjs.io/pt-BR/docs/next/configuration#transform-objectstring-pathtotransformer--pathtotransformer-object)

Reply:
@twistershark Since we are using the `transform`, babel-jest "will load your project's Babel configuration and transform any file matching the /\.[jt]sx?$/ RegExp (in other words, any .js, .jsx, .ts or .tsx file)." However, [testMatch](https://jestjs.io/pt-BR/docs/next/configuration#testmatch-arraystring) by default already have the `spec` as identifier. So to add this will make any different to the code.